packages/material-ui/src/ListItemText/ListItemText.test.js

```diff
@@ -23,102 +26,112 @@ describe('<ListItemText />', () => {
   }));
 
   it('should render with inset class', () => {
-    const wrapper = mount(<ListItemText inset />);
-    const listItemText = findOutermostIntrinsic(wrapper);
-    expect(listItemText.hasClass(classes.inset)).to.equal(true);
-    expect(listItemText.hasClass(classes.root)).to.equal(true);
+    const { container } = render(<ListItemText inset />);
+    expect(container.querySelector('div')).to.have.class(classes.inset);
+    expect(container.querySelector('div')).to.have.class(classes.root);
   });
 
   it('should render with no children', () => {
-    const wrapper = mount(<ListItemText />);
-    const listItemText = findOutermostIntrinsic(wrapper);
-    // wrapper.find('div > *').exists()
-    // https://github.com/airbnb/enzyme/issues/1154
-    expect(listItemText.children().exists()).to.equal(false);
+    const { container } = render(<ListItemText />);
+    expect(container.querySelector('div').querySelectorAll('*').length).to.equal(0);
   });
 
   describe('prop: primary', () => {
     it('should render primary text', () => {
       const ref = React.createRef();
       const text = () => ref.current.textContent;
-      const wrapper = mount(<ListItemText primary="This is the primary text" ref={ref} />);
-      const listItemText = findOutermostIntrinsic(wrapper);
-      const typography = listItemText.find(Typography);
-      expect(typography.exists()).to.equal(true);
-      expect(typography.props().variant).to.equal('body1');
+      const { container } = render(<ListItemText primary="This is the primary text" ref={ref} />);
+      expect(container.querySelectorAll('span.MuiTypography-root').length).to.equal(1);
```

Comment:
Hey @joshwooding why don't use `toHaveLength` for these cases?

Reply:
Hey, it's a current convention. I'm not sure of the reason behind it but maybe @oliviertassinari or @eps1lon can explain :)

Reply:
I'll try to use `to.have.length`. `to.equal` is an escape hatch if you don't have a better matcher. The goal is to have descriptive failure messages.

Reply:
@eps1lon Any preference over `to.have.length` vs `to.have.lengthOf`

Reply:
@joshwooding My perception of the tradeoff so far has been that `predictable test > failure messages`. I believe the quality of the assertions is critical, using a smaller set of assertions (smaller API surface to know) reduces the likeness to use it wrongly which leads to less time wasted than cryptic error messages. Why? Because we run all the assertions, we only debug a dozen, 0.1% of them at a time. There is nothing as predictable as a `===` assertion. I wish https://github.com/power-assert-js/power-assert has taken, but it seems unusable.

Now, the assumption that `predictability > failure messages` is what yields the most saved time is unproven nor a belief shared with most. Over the months, we have start to diverge to the strict usage of `expect(A).to.equal(B)` and grow the assertion API surface we use.

As long as we take extra care of using assertion methods we understand the limits of, I think that we are good.